core/src/main/scala/kafka/server/metadata/BrokerMetadataSnapshotter.scala

```diff
@@ -61,7 +61,7 @@ class BrokerMetadataSnapshotter(
   /**
    * The event queue which runs this listener.
    */
-  val eventQueue = new KafkaEventQueue(time, logContext, threadNamePrefix.getOrElse(""))
+  val eventQueue = new KafkaEventQueue(time, logContext, threadNamePrefix.getOrElse(""), () => new ShutdownEvent())
```

Comment:
Why the supplier here for ShutdownEvent? It looks like KafkaEventQueue just takes in the event object